app/core/DeeplinkManager/CoreLinkNormalizer.test.ts

```diff
@@ -0,0 +1,367 @@
+import { CoreLinkNormalizer } from './CoreLinkNormalizer';
+import { CoreUniversalLink } from './types/CoreUniversalLink';
+import AppConstants from '../AppConstants';
+
+describe('CoreLinkNormalizer', () => {
+  const mockTimestamp = 1234567890;
+
+  beforeEach(() => {
+    jest.spyOn(Date, 'now').mockReturnValue(mockTimestamp);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('normalize', () => {
+    it('normalizes basic metamask:// links', () => {
+      const url = 'metamask://swap';
+      const source = 'test';
+
+      const result = CoreLinkNormalizer.normalize(url, source);
+
+      expect(result.protocol).toBe('metamask');
+      expect(result.action).toBe('swap');
+      expect(result.source).toBe(source);
+      expect(result.isValid).toBe(true);
+      expect(result.isSupportedAction).toBe(true);
+      expect(result.timestamp).toBe(mockTimestamp);
+    });
+
+    it('normalizes metamask:// links with query parameters', () => {
+      const url = 'metamask://swap?from=ETH&to=DAI&amount=100';
+      const source = 'qr-code';
+
+      const result = CoreLinkNormalizer.normalize(url, source);
+
+      expect(result.protocol).toBe('metamask');
+      expect(result.action).toBe('swap');
+      expect(result.params.from).toBe('ETH');
+      expect(result.params.to).toBe('DAI');
+      expect(result.params.amount).toBe('100');
+      expect(result.params.swapPath).toBe('swap?from=ETH&to=DAI&amount=100');
+    });
+
+    it('normalizes https:// universal links', () => {
+      const url = `https://${AppConstants.MM_IO_UNIVERSAL_LINK_HOST}/send`;
+      const source = 'browser';
+
+      const result = CoreLinkNormalizer.normalize(url, source);
+
+      expect(result.protocol).toBe('https');
+      expect(result.action).toBe('send');
+      expect(result.host).toBe(AppConstants.MM_IO_UNIVERSAL_LINK_HOST);
+      expect(result.isValid).toBe(true);
+    });
+
+    it('normalizes universal links with paths and parameters', () => {
+      const url = `https://${AppConstants.MM_IO_UNIVERSAL_LINK_HOST}/dapp/app.uniswap.org?chain=1`;
+      const source = 'deep-link';
+
+      const result = CoreLinkNormalizer.normalize(url, source);
+
+      expect(result.protocol).toBe('https');
+      expect(result.action).toBe('dapp');
+      expect(result.params.chain).toBe('1');
+      expect(result.params.dappPath).toBe('dapp/app.uniswap.org?chain=1');
+    });
+
+    it('converts hr parameter from string to boolean', () => {
+      const url = 'metamask://home?hr=1';
+      const source = 'test';
+
+      const result = CoreLinkNormalizer.normalize(url, source);
+
+      expect(result.params.hr).toBe(true);
+    });
+
+    it('handles hr parameter with value 0', () => {
```

Comment:
S: avoid use of generic meaningless words like handle, manage, ...
```suggestion
    it('uses 0 value hr as a boolean', () => {
```